addrcmd.go

```diff
@@ -55,3 +58,99 @@ func filToEthAddr(cctx *cli.Context) error {
 	fmt.Println(eaddr.String())
 	return nil
 }
+
+func filAddrs(cctx *cli.Context) error {
+	args := cctx.Args()
+	ctx := cctx.Context
+	actorAddrString := args.Get(0)
+
+	expensiveLookup := cctx.Bool("expensive")
+
+	was0xAddr := false
+	var err error
+	var eaddr ethtypes.EthAddress
+	var addr address.Address
+	if actorAddrString[0] == '0' {
+		was0xAddr = true
+		eaddr, err = ethtypes.ParseEthAddress(actorAddrString)
+		if err != nil {
+			return err
+		}
+		addr, err = eaddr.ToFilecoinAddress()
+		if err != nil {
+			return err
+		}
+	} else {
+		addr, err = address.NewFromString(actorAddrString)
+		if err != nil {
+			return err
+		}
+	}
+
+	bg, ts, err := getAnchorPoint(cctx)
+	if err != nil {
+		return err
+	}
+	defer bg.LogStats()
+
+	stateTree, err := lchstate.LoadStateTree(ipldcbor.NewCborStore(bg), ts.ParentState())
+	if err != nil {
+		return err
+	}
+
+	idAddr, err := stateTree.LookupID(addr)
+	if err != nil {
+		return err
+	}
+
+	id, err := address.IDFromAddress(idAddr)
+	if err != nil {
+		return err
+	}
+	eaddr = ethtypes.EthAddressFromActorID(abi.ActorID(id))
+
+	switch p := addr.Protocol(); p {
+	case address.SECP256K1, address.Actor, address.BLS:
+		// If it's an f1/f2/f3 return the f0, and 0x addresses
+		fmt.Println(idAddr)
+		fmt.Println(eaddr)
+	case address.ID:
+		if !was0xAddr {
+			// If it's an f0 return the 0x and if enabled the expensive reverse lookup for the fX address
+			fmt.Println(eaddr)
+		} else {
+			// If it's a masked 0x address, return the f0 and if enabled the expensive reverse lookup for the fX address
+			fmt.Println(addr)
+		}
+
+		if expensiveLookup {
+			var fXAddr address.Address
+			completionErr := fmt.Errorf("query complete")
+			if err := enumActors(ctx, bg, ts, func(actor *types.Actor, addr address.Address) error {
+				if bytes.Equal(addr.Bytes(), idAddr.Bytes()) {
+					fmt.Printf("%+v", actor)
+					//fXAddr = *actor.Address
+					return completionErr
```

Comment:
This lookup function isn't right and *actor.Address can be nil so that doesn't seem to be the answer either

Reply:
Going to remove for now and will add back in, with a followup

Reply:
Note: This was enumerating the wrong HAMT (Actors vs Init), we need to enumerate the Init Actor (f01 as found from the Actors HAMT).